test/Constraints/generics.swift

```diff
@@ -188,7 +188,7 @@ func r22459135() {
 
 // <rdar://problem/19710848> QoI: Friendlier error message for "[] as Set"
 // <rdar://problem/22326930> QoI: "argument for generic parameter 'Element' could not be inferred" lacks context
-_ = [] as Set  // expected-error {{generic parameter 'Element' could not be inferred in cast to 'Set'}} expected-note {{explicitly specify the generic arguments to fix this issue}} {{14-14=<<#Element: Hashable#>>}}
+_ = [] as Set  // expected-error {{using 'Any' as a concrete type conforming to protocol 'Hashable' is not supported}}
```

Comment:
This is more accurate now, awesome!

I know this is an existing diagnostic, but in a subsequent patch, do you mind changing it to not say "not supported". Instead, I would just say "protocol type 'Any' cannot conform to 'Hashable' because only concrete types can conform to protocols" or something like that.

Reply:
Sure! There are currently 2 existing diagnostics - one for objc and one for swift types, should I modify both or just the one related to swift types?

Reply:
I think the objc one already diagnoses the correct cause - static or initializer requirements. The swift one just says “not supported” which suggests it might be supported, but we don’t know if we’re going to implement self conforming existentials yet. Instead of promising a behavior, the diagnostic should explain the language rule - only concrete types can satisfy protocol requirements.